build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildParams.kt

```diff
@@ -308,6 +310,17 @@ val Project.isExperimentalTestFilteringEnabled: Boolean
     get() = systemProperty(TEST_FILTERING_ENABLED).getOrElse("false").toBoolean()
 
 
+val Project.predictiveTestSelectionEnabled: Boolean
+    get() = systemProperty(PREDICTIVE_TEST_SELECTION_ENABLED)
```

Comment:
I think the condition is as before. I don't remember why we enabled test selection even for local builds, though if we already did this I suppose it is fine.

I don't know why you need to evaluate the system property at configuration time. Can you wire up a `Provider<Boolean>` into `predictiveTestSelection.enabled` directly? I suppose that doesn't work when you replace `project.enableExperimentalTestFiltering()` as well.

Another thing which I think would make sense is to disable test selection if `RERUN_ALL_TESTS` is set, right?

Reply:
> I don't remember why we enabled test selection even for local builds, though if we already did this I suppose it is fine.

I wasn't certain, it was enabled previously looking at https://github.com/gradle/gradle/pull/20324. Do you see any harm in enabling locally? It will automatically be disabled if tests are run from IDEA or via `--tests`.

> I don't know why you need to evaluate the system property at configuration time. Can you wire up a Provider<Boolean> into predictiveTestSelection.enabled directly?

Sure, I did it this way for consistency with the other flags in this file but I'll change it.

> I suppose that doesn't work when you replace `project.enableExperimentalTestFiltering()` as well.

What's that?

> Another thing which I think would make sense is to disable test selection if `RERUN_ALL_TESTS` is set, right?

I wasn't aware of that flag. I'll have a look.

Reply:
Resolved by pairing with Stefan.